source/settings/advanced.rst

```diff
@@ -19,6 +19,42 @@ See :ref:`settings` for list of all setting groups.
    :ref:`Login variables <variables-login>` can be used.
 
 
+.. dovecot_core:setting:: mail_cache_max_header_name_length
+   :added: v2.4.0;v3.0.0
+   :default: 100
+   :values: @uint
+
+   Maximum header name length stored in the cache, where 0 stands for unlimited
+   (which is also the former behavior).
+   When enabled, the cache truncates the names to this length in memory and on
+   file. While the header name remains unchanged in the storage, all the headers
+   sharing the first ``mail_cache_max_header_name_length`` prefix characters are
+   de facto aliased and will be considered as the same header on cache fetch.
+
+   Also, attempting to fetch a specific aliased header will succeed even if
+   the header does not actually exist (this does NOT happen when the feature
+   is disable with explicitly with ``mail_cache_max_header_name_length = 0``)
+
+   Example: (``mail_cache_max_header_name_length = 5``)
+   If the mail contains the header ``X-name: value``, attempting to fetch
+   ``X-nam`` or ``X-names`` will also produce ``value`` as result.
```

Comment:
This is related to the `FETCH 1 BODY.PEEK[HEADER.FIELDS ($long)]` question I put to the tester MR. It doesn't just produce "value" with IMAP, but should produce "X-Names: value".